lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -102,6 +101,13 @@
     - ' - C(type) (string): Valid value is C(thin) (default: None).'
     - ' - C(datastore) (string): Datastore to use for the disk. If C(autoselect_datastore) is enabled, filter datastore selection.'
     - ' - C(autoselect_datastore) (bool): select the less used datastore.'
+  cdrom:
```

Comment:
So to me cdrom is part of the hardware, so under the hardware parameter.

Reply:
Then disk should also be moved under hardware... I've followed the same procedure as disk. Adding cdrom into hardware will cause a 3 level dict, which i think is a bit complicated for this. (same for network)

Reply:
You are right, it's not consistent as-is.

@Akasurde What is the direction we are taking this ?

Also, we should also define a sub-spec for the hardware-parameter at some point.

Reply:
I would like to see `cdrom` as separate spec as it is now. Like @pdellaert said, it will complicated if we move it under hardware.

Reply:
Ok, I am confused, about it being a 3rd level dict (and why that even matters) and it being complicated (why ?).

To me putting everything as a top-level parameter is problematic, that's why we introduced the hardware parameter in the first place. The problem here is that if people add it the way they prefer, things will become very inconsistent (and I guess this is why we are having that discussion now, again).